math/src/field/traits.rs

```diff
@@ -282,3 +282,11 @@ pub trait IsPrimeField: IsField {
         Some((x, neg_x))
     }
 }
+
+/// This trait is necessary for sampling a random field element with a uniform distribution.
+pub trait HasDefaultTranscript: IsField {
+    /// This function should truncates the sampled bits to the quantity required to represent the order of the base field
+    /// and returns a field element if the resulting number is less than the modulus, or None if the sampled number
+    /// is greater than the modulus.
+    fn get_random_field_element_from_seed(seed: [u8; 32]) -> Option<FieldElement<Self>>;
```

Comment:
Idea 2: rather than giving a `seed` to this function, you could give it a _seeded RNG_ object from which the implementation can draw as many bytes as needed; this would also remove the `Option<>` from the return type. This would remove the need for Idea 1.

```suggestion
    fn get_random_field_element_from_rng(rng: impl rand::Rng) -> Option<FieldElement<Self>>;
```

With this, the implementation of `get_random_field_element_from_rng()` for u64 fields could do something like
```rust
    let mut sample = [0u8; 8];
    rng.fill(&mut sample);

    mask = /* mask from u46::MAX and MODULUS.leading_zeroes() */

    while sample &= mask >= MODULUS {
        rng.fill(&mut sample)
    }
```

Reply:
As a extra thought from Robin: the trait function signature could also take a mutable reference to the RNG so that, in the Fiat-Shamir transform, the same RNG can be re-used to generate other parts of challenges, e.g., if several field elements are required.
```suggestion
    fn get_random_field_element_from_rng(rng: &mut impl rand::Rng) -> FieldElement<Self>;
```